Seen with tf-head: ModuleNotFoundError: No module named 'keras.optimizers.optimizer_v2'

Problem with tf-head / Keras seen in CI, for instance at https://github.com/horovod/horovod/actions/runs/3656223581/jobs/6180240570

```
___________ ERROR collecting test/parallel/test_tensorflow_keras.py ____________
ImportError while importing test module '/horovod/test/parallel/test_tensorflow_keras.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_tensorflow_keras.py:36: in <module>
    from keras.optimizers.optimizer_v2 import optimizer_v2
E   ModuleNotFoundError: No module named 'keras.optimizers.optimizer_v2'
```
```
___________ ERROR collecting test/parallel/test_tensorflow2_keras.py ___________
ImportError while importing test module '/horovod/test/parallel/test_tensorflow2_keras.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test_tensorflow2_keras.py:35: in <module>
    from keras.optimizers.optimizer_v2 import optimizer_v2
E   ModuleNotFoundError: No module named 'keras.optimizers.optimizer_v2'

```

--- a/test/parallel/test_tensorflow2_keras.py
+++ b/test/parallel/test_tensorflow2_keras.py
@@ -31,8 +31,10 @@ from horovod.common.util import is_version_greater_equal_than
 if is_version_greater_equal_than(tf.__version__, "2.6.0"):
     if version.parse(keras.__version__.replace("-tf", "+tf")) < version.parse("2.9.0"):
         from keras.optimizer_v2 import optimizer_v2
-    else:
+    elif version.parse(keras.__version__.replace("-tf", "+tf")) < version.parse("2.12.0"):
         from keras.optimizers.optimizer_v2 import optimizer_v2
+    else:
+        from keras.optimizers.legacy import optimizer_v2
 else:
     from tensorflow.python.keras.optimizer_v2 import optimizer_v2
 

--- a/test/parallel/test_tensorflow_keras.py
+++ b/test/parallel/test_tensorflow_keras.py
@@ -32,8 +32,10 @@ if is_version_greater_equal_than(tf.__version__, "2.6.0"):
     from keras import backend as K
     if version.parse(keras.__version__.replace("-tf", "+tf")) < version.parse("2.9.0"):
         from keras.optimizer_v2 import optimizer_v2
-    else:
+    elif version.parse(keras.__version__.replace("-tf", "+tf")) < version.parse("2.12.0"):
         from keras.optimizers.optimizer_v2 import optimizer_v2
+    else:
+        from keras.optimizers.legacy import optimizer_v2
 else:
     from tensorflow.python.keras import backend as K
     from tensorflow.python.keras.optimizer_v2 import optimizer_v2

